### PR TITLE
make multicluster secret controller handle updates

### PR DESCRIPTION
**Pull request by ayj**
The multicluster secret controller was silently ignoring updates to
secrets. Users had to delete and re-add secrets if they wanted to
change the credentials for remote clusters. This PR adds update
support and also improves the unit tests.

fixes https://github.com/istio/istio/issues/18708

### CODE REVIEW

**Comment by ramaraochavali** [comment]
LGTM

File: pkg/kube/secretcontroller/secretcontroller_test.go [hunk 1]
```diff
@@ -15,126 +15,168 @@
...
-		t.Fatalf("Unexpected error on secret delete: %v", err)
-	}
+	// Start the secret controller and sleep to allow secret process to start.
+	g.Expect(
+		StartSecretController(clientset, addCallback, updateCallback, deleteCallback, secretNamespace)).
+		Should(Succeed())
 
-	// Test - Verify that the remote controller has been removed.
-	verifyControllerDeleted(t, "delete remote secret controller")
+	time.Sleep(100 * time.Millisecond)
```
Comment (howardjohn) [inline pkg/kube/secretcontroller/secretcontroller_test.go]: Why do we need this?

Comment (ayj) [reply pkg/kube/secretcontroller/secretcontroller_test.go]: Leftover from the previous tests. We don't need it anymore.

**Comment by ayj** [comment]
> Writing `ConfigStoreCache` to `../config/aggregate/fakes/config_store_cache.gen.go`... pilot/pkg/model/config.go:206: running "counterfeiter": exit status 1

TestFixedWindowAlgorithm also failed which seems unrelated. 

/retest

**Comment by howardjohn** [comment]
Its broken by an accidentally reverted revert, should be fixed in ~30 min
or so once CI is done

**Comment by ayj** [comment]
Do I need to rebase, or is the issue in the test infra itself?

**Comment by howardjohn** [comment]
In the infra, waiting then `/retest` will fix it.

You should almost never need to rebase by the way, as the tests will rebase
before running.

**Comment by howardjohn** [comment]
/retest

should be fixed

**Comment by ayj** [comment]
> --- FAIL: TestStreamSecretsUpdateFailures (0.51s)
>     sdsservice_test.go:762: unexpected metric totalPush: expected 3 but got 1

/retest

**Pull request merged**